File: NirvanaSMS.Api.Java/src/main/java/org/nirvanasms/api/core/base/MessageBuilder.java
package org.nirvanasms.api.core.base;

import java.util.Date;

import org.nirvanasms.api.core.interfaces.IBuilder;

public abstract class MessageBuilder<T,T2> extends MessageBase implements IBuilder<T> {
	
	 public abstract T2 validity(int validity);

     public abstract T2 header(String header);

     public abstract T2 sendDateTime(String dateTime, String datePattern);

     public abstract T2 sendDateTime(Date dateTime);
     
	@Override
	public abstract T build();
}

File: NirvanaSMS.Api.CSharp/LibraryProject/Core/Base/MessageBuilder.cs
using NirvanaSMS.Api.Core.Interface;
using System;

namespace NirvanaSMS.Api.Core.Base
{
    public abstract class MessageBuilder<T, T2> : MessageBase, IBuilder<T>
    {
        public abstract T2 SetValidity(int validity);

        public abstract T2 SetHeader(string header);

        public abstract T2 SetSendDateTime(string dateTime);

        public abstract T2 SetSendDateTime(DateTime dateTime);

        public abstract T Build();
    }
}
